### ISSUE TITLE
Updating a disabled user w/o specifying 'enabled' property enables the user

### ISSUE THREAD

**Issue by BigPandaToo** [elastic/elasticsearch#67533]
When updating a disabled user it looks like we inadvertently enale it as 'enabled' proberty is being set to 'true' by default regardless of current value.
Repro:
# Create disabled user.
POST /_security/user/test
{
  "password": "changeme",
  "roles": ["superuser"],
  "enabled": false
}
# Verify user is disabled.
GET /_security/user/test
# Update user, not specifying the `enabled` property.
POST /_security/user/test
{
  "roles": ["superuser"],
  "email": "user@example.com"
}
# Retrieve user, notice they have been enabled.
GET /_security/user/test

**Comment by albertzaharovits** [comment]
This behavior exists for a very long time (2358309), and I agree it is unexpected. I think we should fix it.

**Comment by tvernum** [comment]
I disagree.
`enabled` is no different than any other property on a user, with the 1 exception of `password` (for which we have explicit handling).

For example:

Create a user with several fields populated:
```
POST /_security/user/example
{
   "full_name": "Tony Stark",
   "metadata": { "alias": "ironman" },
   "password": "password",
   "enabled": true,
   "roles": [ "superuser" ]
}
===
{
  "created": true
}

GET /_security/user/example
===
{
  "example": {
    "username": "example",
    "roles": [
      "superuser"
    ],
    "full_name": "Tony Stark",
    "email": null,
    "metadata": {
      "alias": "ironman"
    },
    "enabled": true
  }
}
```

Then update a user, but do not specify the already populated fields (other than `roles` which is always required)  
```
POST /_security/user/example
{
  "email": "tony@stark.co", "roles": [ ]
}
===
{
  "created": false
}
```

As we can see, everything (`full_name`, `metadata` and `enabled`) is reset to its default values. The `POST` is, by design, an alias for `PUT` - it is explicitly not a `PATCH`.
```
GET /_security/user/example
===
{
  "example": {
    "username": "example",
    "roles": [],
    "full_name": null,
    "email": "tony@stark.co",
    "metadata": {},
    "enabled": true
  }
}
```

I guess we could decide that we want to support `PATCH` on security entities, but right now we don't - a `PUT` is a complete PUT.

**Comment by BigPandaToo** [comment]
I understand the behaviour, but I think silently enabling explicetly disabled user may pose a security vulnerability. We can discuss to see if we want to change it.

**Comment by albertzaharovits** [comment]
I looked at the other request fields too before commenting.
Although I see your point Tim, I think the `roles` and `password` fields are cheating a bit. As you also describe, we don't update the user with an empty password if the update request doesn't contain a password because that would break the min password length requirement, and we've marked the `roles` field as required, even though it's reasonable to be able to update a user without changing its roles. I think these exceptions dilute the PUT/PATCH principle, to the point that `enable` can be made another such exception. It's a balancing act, it might be trappy to enable a disabled user when updating, but with every exception we create a precedent for the other fields, which are potentially used in role mappings, and the API becomes PATCH instead of PUT, like the other APIs.

I slightly maintain my viewpoint that `enabled` shouldn't work like it does, it should be an exception like `password`, but in cases like this, given the status quo, the best option that minimizes failed user expectations might be to change nothing and enshrine the behavior in the docs?

**Comment by bytebilly** [comment]
@albertzaharovits I agree with your suggestion, even because both behaviors may be confusing and switching (with a breaking change) could be even more confusing.

+1 to update documentation and clarify!
